federation-2/router-bridge/js-src/plan.ts

```diff
@@ -19,65 +32,329 @@ export class BridgeQueryPlanner {
     this.planner = new QueryPlanner(this.composedSchema);
   }
 
-  plan(
-    operationString: string,
-    operationName?: string
-  ): ExecutionResult<QueryPlan> {
+  plan(operationString: string, operationName?: string): PlannerResult {
+    let operationDocument: DocumentNode;
     try {
-      const operationDocument = parse(operationString);
-
-      // Federation does some validation, but not all.  We need to do
-      // all default validations that are provided by GraphQL.
-      const validationErrors = validate(this.apiSchema, operationDocument);
-      if (validationErrors.length > 0) {
-        throw validationErrors;
-      }
+      operationDocument = parse(operationString);
+    } catch (parseError) {
+      return {
+        usageReportingSignature: "## GraphQLParseFailure\n",
+        errors: [parseError],
+      };
+    }
+    // Federation does some validation, but not all.  We need to do
+    // all default validations that are provided by GraphQL.
+    const validationErrors = validate(this.apiSchema, operationDocument);
+    if (validationErrors.length > 0) {
+      return {
+        usageReportingSignature: "## GraphQLValidationFailure\n",
+        errors: validationErrors,
+      };
+    }
 
-      const operation = operationFromDocument(
+    // This throws GraphQLErrors, among which are `GraphQLUnknownOperationName`
+    let operation: Operation;
+    try {
+      operation = operationFromDocument(
         this.composedSchema,
         operationDocument,
         operationName
       );
-
-      return { data: this.planner.buildQueryPlan(operation) };
     } catch (e) {
-      const errors = Array.isArray(e) ? e : [e];
-      return { errors };
+      const usageReportingSignature = e.message.startsWith(
+        "Unknown operation named"
+      )
+        ? "## GraphQLUnknownOperationName\n"
+        : "## GraphQLValidationFailure\n";
+      return {
+        usageReportingSignature,
+        errors: [e],
+      };
     }
+
+    // I double checked, this function doesn't throw
+    const usageReportingSignature = defaultUsageReportingSignature(
+      operationDocument,
+      operationName
+    );
+
+    return {
+      data: this.planner.buildQueryPlan(operation),
+      usageReportingSignature,
+    };
   }
 }
 
 export function queryPlanner(schemaString: string): BridgeQueryPlanner {
   return new BridgeQueryPlanner(schemaString);
 }
 
-export function plan(
-  schemaString: string,
-  operationString: string,
-  operationName?: string
-): ExecutionResult<QueryPlan> {
-  try {
-    const composedSchema = buildSchema(schemaString);
-    const apiSchema = composedSchema.toAPISchema();
-    const operationDocument = parse(operationString);
-    const graphqlJsSchema = apiSchema.toGraphQLJSSchema();
+// ---------------------
```

Comment:
you presumably want some sort of comment saying that this is copy-and-pasted from a particular file in apollo-server and that we're planning to refactor to combine them later.